lib/dynamoid/persistence.rb

```diff
@@ -801,9 +801,20 @@ def delete
 
       @destroyed = true
 
-      Dynamoid.adapter.delete(self.class.table_name, hash_key, options)
+      Dynamoid.adapter.delete(self.class.table_name, hash_key, options).tap { update_asociations }
```

Comment:
**minor:** TBH I would avoid introducing new method (`update_asociations`) which can conflict with user defined methods in model and would inline it into the `delete` method.